extensions/kinesis/src/main/java/com/hazelcast/jet/kinesis/impl/sink/KinesisSinkP.java

```diff
@@ -170,10 +170,7 @@ public boolean complete() {
 
     @Override
     public boolean saveToSnapshot() {
-        if (sendResult != null) {
-            checkIfSendingFinished();
-        }
-        return sendResult == null;
+        return complete();
```

Comment:
In the complete() method 
```
 if (sendResult != null) {
   ...
 }
 if (sendResult == null) {
  ...
}
````
can be changed to if/else. Because if (sendResult == null) will always be true

Reply:
No, because the `checkIfSendingFinished` can nullify the `sendResult`

Reply:
Then maybe add a comment ? . Because it is difficult to see it